### PR TITLE
Creates the sample Admin Node.js session management quickstart app.

### PR DESCRIPTION
**Pull request by bojeil-google**
This app depends on the upcoming release of the session management API
in firebase-admin version 5.12.0.

### CODE REVIEW

File: auth-sessions/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+## Developer Setup
+
+### Dependencies
+
+To set up a development environment to build the sample from source, you must
+have the following installed:
+- Node.js (>= 6.0.0)
+- npm (should be included with Node.js)
+
+Download the Firebase source and its dependencies with:
```
Comment (hiranya911) [inline auth-sessions/README.md]: s/Firebase/sample application/

File: auth-sessions/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+var config = {
+  apiKey: "...",
+  authDomain: "...",
+  databaseURL: "...",
+  storageBucket: "...",
+  messagingSenderId: "..."
+};
+module.exports = config;
+```
+Copy and paste the Web snippet code configuration found in the console (either by clicking "Add
```
Comment (hiranya911) [inline auth-sessions/README.md]: I had to read this a couple of times to get the picture. Consider turning the text in the parentheses into separate sentences or bullet points.

File: auth-sessions/app.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  return admin.auth().getUser(decodedClaims.sub).then(function(userRecord) {
+    var html = '<!DOCTYPE html>' +
+      '<html>' +
+      '<meta charset="UTF-8">' +
+      '<link href="style.css" rel="stylesheet" type="text/css" media="screen" />' +
+      '<meta name="viewport" content="width=device-width, initial-scale=1">' +
+      '<title>Sample Profile Page</title>' +
+      '<body>' +
+      '<div id="container">' +
+      '  <h3>Welcome to My Awesome App, '+( userRecord.displayName || 'N/A') +'</h3>' +
```
Comment (hiranya911) [inline auth-sessions/app.js]: Session Management Example App?

File: auth-sessions/app.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+ * @param {string} cookie The CSRF token name.
+ * @param {string} value The CSRF token value to save.
+ * @return {function} The middleware function to run.
+ */
+function attachCsrfToken(url, cookie, value) {
+  return function(req, res, next) {
+    if (req.url == url) {
+      res.cookie(cookie, value);
+    }
+    next();
```
Comment (hiranya911) [inline auth-sessions/app.js]: I can't find the definition of this function. Is this an Express built-in?

Comment (bojeil-google) [reply auth-sessions/app.js]: https://expressjs.com/en/guide/writing-middleware.html

File: auth-sessions/README.md (hunk 1 above)
Comment (bojeil-google) [reply auth-sessions/README.md]: Done

File: auth-sessions/README.md (hunk 2 above)
Comment (bojeil-google) [reply auth-sessions/README.md]: Yeah, it was confusing. I have revised this paragraph.

File: auth-sessions/app.js (hunk 3 above)
Comment (bojeil-google) [reply auth-sessions/app.js]: Done

File: auth-sessions/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+To set up a development environment to build the sample from source, you must
+have the following installed:
+- Node.js (>= 6.0.0)
+- npm (should be included with Node.js)
+
+Download the sample application source and its dependencies with:
+
+```bash
+git clone https://github.com/firebase/quickstart-nodejs.git
```
Comment (nicolasgarnier) [inline auth-sessions/README.md]: You can drop the `.git`

Comment (bojeil-google) [reply auth-sessions/README.md]: Done

File: auth-sessions/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  apiKey: "...",
+  authDomain: "...",
+  databaseURL: "...",
+  storageBucket: "...",
+  messagingSenderId: "..."
+};
+module.exports = config;
+```
+Copy and paste the Web snippet code configuration found in the console to the `config.js` file.
+You can find the snippet either by clicking the "Web setup" button in the Firebase Console
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply auth-sessions/README.md]: Ok added more instructions and links to setup instructions.

File: auth-sessions/README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+```bash
+git clone https://github.com/firebase/quickstart-nodejs.git
+cd quickstart-nodejs/auth-sessions
+npm install
+```
+
+### Configuring the app
+
+In the Firebase Console, make sure Email/Password and Google providers are enabled.
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply auth-sessions/README.md]: Ok added more instructions and links to setup instructions.

**Pull request merged**